### PR TITLE
Implement list() method causal manager

### PR DESCRIPTION
**Pull request by gaugup**
## Description

Implement list() method causal manager

## Areas changed

npm packages changed:

- [ ] responsibleai/causality
- [ ] responsibleai/core-ui
- [ ] responsibleai/counterfactuals
- [ ] responsibleai/dataset-explorer
- [ ] responsibleai/fairness
- [ ] responsibleai/interpret
- [ ] responsibleai/localization
- [ ] responsibleai/mlchartlib
- [ ] responsibleai/model-assessment

Python packages changed:

- [ ] raiwidgets
- [x] responsibleai
- [ ] erroranalysis
- [ ] rai_core_flask

## Tests

- [ ] No new tests required.
- [x] New tests for the added feature are part of this PR.
- [ ] I validated the changes manually.

## Documentation:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: responsibleai/tests/causal_manager_validator.py [hunk 1]
```diff
@@ -57,6 +59,9 @@ def validate_causal(rai_insights, data, target_column,
     assert len(results) == 1
     _check_causal_result(results[0])
 
+    _check_causal_properties(rai_insights.causal.list(),
+                             expected_causal_effects=1)
```
Comment (riedgar-ms) [inline responsibleai/tests/causal_manager_validator.py:63]: Is there a test case for when nothing is supposed to be there?

File: responsibleai/responsibleai/managers/causal_manager.py [hunk 2]
```diff
@@ -328,7 +329,25 @@ def get(self):
...
+        """List information about the CausalManager.
+
+        :return: A dictionary of properties.
+        :rtype: Dict
+        """
+        props = {ListProperties.MANAGER_TYPE: self.name}
+        causal_props_list = []
+        for result in self._results:
+            causal_config_dict = result.config.get_config_as_dict()
+            causal_config_dict['global_effects_computed'] = \
```
Comment (imatiach-msft) [inline responsibleai/responsibleai/managers/causal_manager.py:341]: I think these should be constants? either in this file or in some global constants file.

File: responsibleai/responsibleai/managers/causal_manager.py [hunk 3]
```diff
@@ -328,7 +329,25 @@ def get(self):
         return self._results
 
     def list(self):
-        pass
+        """List information about the CausalManager.
+
+        :return: A dictionary of properties.
+        :rtype: Dict
```
Comment (imatiach-msft) [inline responsibleai/responsibleai/managers/causal_manager.py:335]: nit: lowercase dict

**Review by imatiach-msft — APPROVED** [review]

File: responsibleai/tests/causal_manager_validator.py (hunk 1 above)
Comment (gaugup) [reply responsibleai/tests/causal_manager_validator.py:63]: Done. Added the checking of properties when nothing is computed.

File: responsibleai/responsibleai/managers/causal_manager.py (hunk 3 above)
Comment (gaugup) [reply responsibleai/responsibleai/managers/causal_manager.py:335]: Done

File: responsibleai/responsibleai/managers/causal_manager.py (hunk 2 above)
Comment (gaugup) [reply responsibleai/responsibleai/managers/causal_manager.py:341]: Done

**Review by riedgar-ms — APPROVED** [review]

**Pull request merged**